outlines/processors/base_logits_processor.py

```diff
@@ -14,10 +14,11 @@ def is_mlx_array(logits):
     return isinstance(logits, mx.array)
 
 
-class BaseLogitsProcessor(Protocol):
+class OutlinesLogitsProcessor(Protocol):
```

Comment:
Why not call it `LogitsProcessor`?

Reply:
Actually, do we really need that class if all logit processors are ever going to be `FSMLogitsProcessors`

Reply:
My thinking here was if we want a new logits processor which isn't for structured generation, e.g. `RepetitionPenaltyLogitsProcessor`. We can easily implement this processor once and use it in every `outlines.models` without any changes specific to the model.

This provides us the flexiiblity to have the inference library handle the decoder pass and sampling, and `outlines` can handle all logits augmentation.

Another example: `mlxlm` doesn't support `stop_strings` and `stop_strings` is broken in the latest `transformers` version (`4.41.2`). We could implement a `StopStringsLogitsProcessor` and make it available to all inference libraries. Not arguing that this should be a priority, but I am arguing that this provides us flexibility and opportunities.